### PR TITLE
Library for reading from resistive touchscreens

### PR DESCRIPTION
**Pull request by ladyada**
using just digital/analog i/o for lowest cost connection

### CODE REVIEW

**Comment by ladyada** [comment]
@jerryneedell @atmakersorg @tg-techie hiya wanna try this library?

**Comment by jerryneedell** [comment]
sure -- looks like fun

**Comment by jerryneedell** [comment]
hooked up a PID333 https://www.adafruit.com/product/333 -- it's alive !
```
Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 4.0.0-beta.2-23-g72f0783b6 on 2019-02-07; Adafruit Feather M4 Express with samd51j19
>>>
>>> import touchscreen_simpletest
(23, 21, 19871)
(22, 21, 29087)
(23, 21, 31935)
(23, 21, 33471)
(23, 21, 33695)
(23, 21, 34207)
(23, 21, 33823)
(23, 21, 33631)
(23, 21, 33983)
(23, 21, 33695)
(23, 21, 33407)
(23, 21, 32831)
(23, 21, 32479)
(23, 21, 32223)
(23, 21, 31231)
(23, 21, 30047)
(23, 21, 28095)
(23, 21, 23871)
```
now to figure out  whats x and whats y....

**Comment by TG-Techie** [comment]
Sure. I'll test it when having access to a resistive touch screen / pyportal

**Comment by jerryneedell** [comment]
It says x2/y2 must be analog capable, but I get this error: 
```
Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 4.0.0-beta.2-23-g72f0783b6 on 2019-02-07; Adafruit Feather M4 Express with samd51j19
>>> 
>>> 
>>> import touchscreen_simpletest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "touchscreen_simpletest.py", line 14, in <module>
  File "adafruit_touchscreen.py", line 138, in touch_point
  File "adafruit_touchscreen.py", line 138, in touch_point
  File "adafruit_touchscreen.py", line 135, in touch_point
ValueError: Pin does not have ADC capabilities
>>> 
```
with this test:
```
import board
import adafruit_touchscreen



# These pins are used as both analog and digital! XR and YU must be analog
# and digital capable. XL and YD just need to be digital
ts = adafruit_touchscreen.Touchscreen(board.D5, board.A4,
                                      board.D6, board.A1,
                                      calibration=((5200, 59000), (5800, 57000)),
                                      size=(320, 240))

while True:
    p = ts.touch_point
    if p:
        print(p)

```
It works if I use A1-A4

**Comment by TG-Techie** [comment]
it think it will work well with:
https://github.com/TG-Techie/tg_gui/blob/master/_touch.py
I'm excited to try

**Comment by ladyada** [comment]
updated comments to clarify x- must also be analog (we use it for the pressure sensing

**Comment by jerryneedell** [comment]
I'm having some difficulty convincing myself the pins are handled correctly, but I'm not sure I have very clear documentation for the pins on my screen.  https://www.adafruit.com/product/333
I can make my screen work by trial and error.

Are you convinced the pin mapping is correct? should I put more time into trying to reconcile it or just make it work? 

Once I get them connected in a way that makes sense, the library seems to be working fine.

**Comment by ladyada** [comment]
it doesnt actually matter which pins you use as long as they're connected in pairs. if you flip them around its the same as rotating the screen.

**Comment by ladyada** [comment]
@jerryneedell zis good to merge? if so plz do a review :)

**Review by jerryneedell — APPROVED** [review]
Looks good to me. I tested it with a touchscreen and once I got the pins in the right order, it seems to work well. Lots more to explore as far as function, like using the calibrations etc, but I think this is ok to merge and get out for wider test.

File: README.rst [hunk 1]
```diff
@@ -57,7 +53,20 @@ To install in a virtual environment in your current project:
...
 =============
 
-.. todo:: Add a quick, simple example. It and other examples should live in the examples folder and be included in docs/examples.rst.
+.. code-block:: python
+	import board
+	import adafruit_touchscreen
+
+	# These pins are used as both analog and digital! XR and YU must be analog
+	# and digital capable. XL and YD just need to be digital
+	ts = adafruit_touchscreen.Touchscreen(board.TOUCH_XL, board.TOUCH_XR,
```
Comment (jerryneedell) [inline README.rst:64]: Just noticed this, it should be updated to require XR be analog as well

Comment (ladyada) [reply README.rst:64]: fixed

**Pull request merged**